assets/src/edit-story/components/fontPicker/pickerContainer.js

```diff
@@ -0,0 +1,307 @@
+            name.toLowerCase().startsWith(searchValue.toLowerCase())
+          )
+        : fonts,
+    [searchValue, fonts]
+  );
+
+  // Font options that include search term but not start with
+  const includeSearchFonts = useMemo(
+    () =>
+      searchValue && searchValue !== ''
+        ? fonts.filter(
+            ({ name }) =>
+              name.toLowerCase().indexOf(searchValue.toLowerCase()) > 0
+          )
+        : [],
+    [searchValue, fonts]
+  );
+
+  useEffect(() => {
+    const combinedFontList = []
+      .concat(recentUsedFonts)
+      .concat(normalFonts)
+      .concat(includeSearchFonts)
+      .slice(currentIndex, currentIndex + 10)
+      .map(({ name }) => name);
+
+    if (combinedFontList.length > 0) {
+      getMenuFonts(combinedFontList).then((result) => {
+        const resultArray = result.replace(/}/g, '}},').split('},');
+        setMenuFonts(resultArray);
+      });
+    }
+  }, [
+    currentIndex,
+    getMenuFonts,
+    recentUsedFonts,
+    normalFonts,
+    includeSearchFonts,
+  ]);
+
+  useEffect(() => {
+    inputRef.current.focus();
+  });
+
+  useEffect(() => {
+    const listElement = listContainerRef.current;
+
+    const handleScroll = () => {
+      let { scrollTop } = listElement;
+      scrollTop -= 5;
+      let currentVisibleIndex = Math.floor(scrollTop / 34);
+      if (
+        recentUsedFonts.length > 0 &&
+        currentVisibleIndex > recentUsedFonts.length
+      ) {
+        scrollTop -= 10;
+        currentVisibleIndex =
+          Math.floor(scrollTop / 34) - recentUsedFonts.length;
+      }
+      if (normalFonts.length > 0 && currentVisibleIndex > normalFonts.length) {
+        scrollTop -= 10;
+      }
+      currentVisibleIndex = Math.floor(scrollTop / 34);
+      if (currentVisibleIndex < 0) {
+        currentVisibleIndex = 0;
+      }
+      if (currentVisibleIndex !== currentActiveRef.current) {
+        currentActiveRef.current = currentVisibleIndex;
+        setCurrentIndex(currentVisibleIndex);
+      }
+    };
+    listElement.addEventListener('scroll', handleScroll, { passive: false });
+
+    return () => {
+      listElement.removeEventListener('scroll', handleScroll);
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useFocusOut(pickerContainerRef, toggleOptions);
+
+  useKeyDownEffect(pickerContainerRef, { key: 'esc' }, toggleOptions, [
+    toggleOptions,
+  ]);
+
+  const handleItemClick = (option, slug) => {
+    handleCurrentValue(option);
+    addUsedFont(slug);
+  };
+
+  const renderListWithOptions = (options, offset = 0) => {
+    if (options?.length > 0) {
+      return (
+        <List aria-multiselectable={false} aria-required={false} ref={listRef}>
+          {options.map(({ name, value, slug }, index) => (
+            <Item
+              key={value}
+              onClick={() => handleItemClick(value, slug)}
+              fontFamily={
+                index < currentIndex + 10 - offset &&
```

Comment:
I don't quite understand this expression. Isn't `name` always applicable here?

Reply:
Correct. All the options has it's font family so this expression not necessary.